internal/util/download.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+// GetResponse returns the response of a given URL
+func GetResponse(url string) (*http.Response, error) {
```

Comment:
I recommend follow https://github.com/tidbcloud/tidbcloud-cli/blob/18d8da1b01908bd2b0c876f05d25fe2730a729d1/internal/service/aws/s3/uploader.go#L164 to enable debug.

Reply:
I also suggest adding context to this function. Currently although the cancel works fine(since the main goroutine receives `InterruptError` and quit all goroutines. And the cancel not work in withoutprompt.